File: javaDemo v1.0/src/me/jenny/PO/DashPage.java
package me.jenny.PO;

import me.jenny.Public.LoggerControler;
import io.appium.java_client.android.AndroidDriver;
import org.apache.poi.openxml4j.exceptions.InvalidFormatException;
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.testng.Assert;

import java.io.IOException;

public class DashPage extends BasePage {

	/**
	 * 页面元素
	 */

	//首页搜索框
	public static final By search_loc = By.id(BaseId + "search_button");

	//点击搜索框弹出的输入框
	public static final By input_loc = By.id(BaseId + "search_src_text");

	//搜索结果中的第一个结果：菜谱集
	public static final By clloection_loc = By.id(BaseId+"collect_item");

	//收藏菜谱集按钮
	public static final By favorite_loc = By.id(BaseId + "action_fav");

	//fail_fav_loc = (By.NAME,"取消收藏成功")

	//收藏菜谱当前页面的菜名
	public static final By title_loc = By.id(BaseId+"title");

	/**
	 * 操作方法
	 */
	//定位点击搜索框，进入搜索页面
	public void ClickSearchBox(){
		this.clickEle(search_loc);
	}

	//定位输入框，输入菜谱名
	public void InputRecipe(String recipe){
		this.sendKeys(input_loc,recipe);
	}

	//点击菜谱集（搜索结果中的的第一项）
	public void ClickCollectionItem(){
		this.clickEle(clloection_loc);
	}

	//点击收藏
	public void ClickFavorite(){
		this.clickEle(favorite_loc);
	}

	//收藏菜谱系列名
	public String GetRecipeName(){
		WebElement ele = this.findEle(title_loc);
		return ele.getText();
	}

	/**
	 * 业务封装
	 */

	//搜索 - 收藏
	public void Search(String recipe,Class name) throws IOException, InvalidFormatException {
		LoggerControler log = this.MyLogger(name);
		AndroidDriver driver = super.SetDriver();
		this.ClickSearchBox();
		this.InputRecipe(recipe);
		log.info("本次搜索的食材是："+recipe);
		driver.sendKeyEvent(66);
		this.ClickCollectionItem();
		//this.ClickFavorite();
		log.info("收藏的菜谱名为："+this.GetRecipeName());
		super.captureScreen("Search_"+recipe);
		Assert.assertTrue(GetRecipeName().contains(recipe));
		log.info("该Case测试通过");
	}
}

File: pythonDemo v1.0/PO/DashPage.py
#coding=utf-8
__author__ = 'JennyHui'
from appium import webdriver
from selenium.webdriver.common.by import By
import BasePage

'''
好豆菜谱首页涉及的所有页面元素 - 操作方法 -> 封装
'''
driver = webdriver.Remote('http://localhost:4723/wd/hub', BasePage.Base.capabilities)

class Dash(BasePage.Base):

	#搜索框
	search_loc = (By.ID,"com.haodou.recipe:id/search_button")
	#输入框
	input_loc = (By.ID,"com.haodou.recipe:id/search_src_text")

	#菜谱集
	clloect_loc =(By.ID,"com.haodou.recipe:id/collect_item")

	#收藏按钮
	favorite_loc = (By.ID,"com.haodou.recipe:id/action_fav")
	fail_fav_loc = (By.NAME,"取消收藏成功")

	#菜名
	title_loc = (By.ID,"com.haodou.recipe:id/title")

	#定位点击搜索框，进入搜索页面
	def click_search_box(self):
		self.clickButton(self.search_loc)

	#定位输入框，输入菜谱名
	def input_recipe(self,recipe):
		self.send_keys(self.input_loc,recipe)

	#点击菜谱集（搜索后的第一项）
	def click_clloect_item(self):
		self.clickButton(self.clloect_loc)

	#点击收藏
	def click_favorite_clloect(self):
		self.clickButton(self.favorite_loc)

	#收藏的菜谱系列名
	def get_recipe_name(self):
		els = self.find_elements(self.title_loc)
		#recipe_list = json.dumps(list, encoding='UTF-8', ensure_ascii=False)
		print u'收藏的菜谱名:'+els[0].text

#搜索/收藏
def search(recipe):
	dash_page = Dash(driver)
	dash_page.click_search_box()
	print u'搜索关键字:'+recipe
	dash_page.input_recipe(recipe)
	driver.press_keycode(66)
	dash_page.click_clloect_item()
	dash_page.click_favorite_clloect()
	dash_page.get_recipe_name()
	dash_page.saveScreenshot('search_'+recipe)
	driver.quit()



